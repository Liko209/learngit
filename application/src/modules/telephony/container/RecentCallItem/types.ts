/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-06-27 14:16:31
 * Copyright Â© RingCentral. All rights reserved.
 */
import { Caller } from 'sdk/module/RCItems/types';
import { RCMessage } from 'sdk/module/RCItems';

type Props = {
  id: string;
  selected: boolean;
  handleClick: (focusIndex?: number) => void;
};

type ViewProps = {
  caller?: Caller;
  icon: string;
  startTime: string;
  isMissedCall: boolean;
  direction: RCMessage['direction'];
  selected?: boolean;
  makeCall: (event: React.MouseEvent) => void;
};

export { Props, ViewProps };
