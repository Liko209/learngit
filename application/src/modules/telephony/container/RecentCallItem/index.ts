/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-06-27 14:16:31
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './RecentCallItem';
