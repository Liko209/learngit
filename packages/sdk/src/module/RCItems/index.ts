/*
 * @Author: Rito.Xiao (rito.xiao@ringcentral.com)
 * @Date: 2019-05-21 19:19:08
 * Copyright Â© RingCentral. All rights reserved.
 */

export { RCMessage, Attachment } from './types';
export { CALL_DIRECTION, RC_MESSAGE_TYPE } from './constants';
