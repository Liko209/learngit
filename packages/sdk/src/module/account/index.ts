/*
 * @Author: Jerry Cai (jerry.cai@ringcentral.com)
 * @Date: 2019-04-04 10:15:25
 * Copyright Â© RingCentral. All rights reserved.
 */

export { AccountService } from './service/NewAccountService';
