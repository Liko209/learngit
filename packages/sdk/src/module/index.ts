/*
 * @Author: Thomas thomas.yang@ringcentral.com
 * @Date: 2018-12-26 15:14:30
 * Copyright Â© RingCentral. All rights reserved.
 */

export { default as ProgressService, PROGRESS_STATUS } from './progress';
export { default as GroupService } from './group';
