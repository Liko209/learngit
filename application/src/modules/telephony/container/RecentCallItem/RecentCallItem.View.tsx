/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-06-27 14:16:31
 * Copyright Â© RingCentral. All rights reserved.
 */
import React from 'react';
import { observer } from 'mobx-react';
import {
  JuiRecentCallItem,
  StyledContactWrapper,
  StyledCallLogStatusWrapper,
  StyledTime,
} from 'jui/pattern/Dialer';
import { ContactInfo } from '@/modules/phone/container/ContactInfo';
import { ViewProps } from './types';
import { JuiIconography, IconColor } from 'jui/foundation/Iconography';

@observer
class RecentCallItemView extends React.Component<ViewProps> {
  render() {
    const {
      caller,
      icon,
      isMissedCall,
      startTime,
      direction,
      handleClick,
      selected,
    } = this.props;
    const color: IconColor = isMissedCall
      ? ['accent', 'tomato']
      : ['grey', '600'];
    return (
      <JuiRecentCallItem onClick={handleClick} tabIndex={0} selected={selected}>
        <StyledContactWrapper>
          <ContactInfo
            caller={caller}
            isMissedCall={isMissedCall}
            direction={direction}
            disableOpenMiniProfile
          />
        </StyledContactWrapper>
        <StyledCallLogStatusWrapper>
          <JuiIconography iconSize='medium' iconColor={color}>
            {icon}
          </JuiIconography>
        </StyledCallLogStatusWrapper>
        <StyledTime>{startTime}</StyledTime>
      </JuiRecentCallItem>
    );
  }
}

export { RecentCallItemView };
