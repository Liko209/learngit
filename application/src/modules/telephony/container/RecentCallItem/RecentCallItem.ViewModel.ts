/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-06-27 14:16:31
 * Copyright Â© RingCentral. All rights reserved.
 */

import { computed, action } from 'mobx';
import { ENTITY_NAME } from '@/store';
import { getEntity } from '@/store/utils';
import { StoreViewModel } from '@/store/ViewModel';
import { Props, ViewProps } from './types';
import CallLogModel from '@/store/models/CallLog';
import { CALL_RESULT } from 'sdk/module/RCItems/callLog/constants';
import { CALL_DIRECTION } from 'sdk/module/RCItems';
import { dialerTimestamp } from '@/utils/date';
import { CallLog } from 'sdk/module/RCItems/callLog/entity';

class RecentCallItemViewModel extends StoreViewModel<Props>
  implements ViewProps {
  @computed
  get data() {
    return getEntity<CallLog, CallLogModel, string>(
      ENTITY_NAME.CALL_LOG,
      this.props.id,
    ) as CallLogModel;
  }

  @computed
  get icon() {
    if (this.isMissedCall) {
      return 'missedcall';
    }
    const { direction } = this.data;
    return direction === CALL_DIRECTION.INBOUND ? 'incall' : 'outcall';
  }

  @computed
  get isMissedCall() {
    const { result } = this.data;
    return result === CALL_RESULT.MISSED || result === CALL_RESULT.VOICEMAIL;
  }

  @computed
  get direction() {
    return this.data.direction;
  }

  @computed
  get startTime() {
    return dialerTimestamp(this.data.startTime);
  }

  @computed
  get caller() {
    const { direction } = this.data;
    return direction === CALL_DIRECTION.INBOUND ? this.data.from : this.data.to;
  }

  @action
  makeCall = (event: React.MouseEvent) => {
    const { handleClick } = this.props;
    event.stopPropagation();

    handleClick();
  }
}

export { RecentCallItemViewModel };
