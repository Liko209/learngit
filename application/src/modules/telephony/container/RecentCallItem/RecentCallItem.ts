/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-06-27 14:16:31
 * Copyright Â© RingCentral. All rights reserved.
 */

import { buildContainer } from '@/base';
import { RecentCallItemView } from './RecentCallItem.View';
import { RecentCallItemViewModel } from './RecentCallItem.ViewModel';
import { Props } from './types';

const RecentCallItem = buildContainer<Props>({
  View: RecentCallItemView,
  ViewModel: RecentCallItemViewModel,
});

export { RecentCallItem, Props };
