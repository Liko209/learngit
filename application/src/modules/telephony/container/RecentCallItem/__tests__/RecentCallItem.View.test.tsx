/*
 * @Author: Shining Miao (shining.miao@ringcentral.com)
 * @Date: 2019-06-28 13:47:39
 * Copyright Â© RingCentral. All rights reserved.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { RecentCallItemView } from '../RecentCallItem.View';
import { CALL_DIRECTION } from 'sdk/module/RCItems';

describe('RecentCallItemView', () => {
  describe('render', () => {
    beforeEach(() => {
      jest.resetAllMocks();
    });
    it('should render each record item [JPT-2316, JPT-2317]', () => {
      const props = {
        caller: {
          phoneNumber: '00000',
          location: 'abc',
          name: 'Shining',
        },
        icon: 'missedcall',
        isMissedCall: true,
        startTime: '2019-06-28T02:09:28.595Z',
        direction: CALL_DIRECTION.INBOUND,
        handleClick: () => {},
        isTransferPage: false,
      };
      const Wrapper = shallow(<RecentCallItemView {...props} />);
      expect(Wrapper).toMatchSnapshot();
    });
  });
});
