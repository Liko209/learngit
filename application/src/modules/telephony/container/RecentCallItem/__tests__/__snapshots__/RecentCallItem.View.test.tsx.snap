// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentCallItemView render should render each record item [JPT-2316, JPT-2317] 1`] = `
<Styled(JuiListItem)
  tabIndex={0}
>
  <styled.div>
    <Container(ContactInfoView)
      caller={
        Object {
          "location": "abc",
          "name": "Shining",
          "phoneNumber": "00000",
        }
      }
      direction="Inbound"
      disableOpenMiniProfile={true}
      isMissedCall={true}
    />
  </styled.div>
  <styled.div>
    <ForwardRef(JuiIconography)
      iconColor={
        Array [
          "accent",
          "tomato",
        ]
      }
      iconSize="medium"
    >
      missedcall
    </ForwardRef(JuiIconography)>
  </styled.div>
  <styled.div>
    2019-06-28T02:09:28.595Z
  </styled.div>
</Styled(JuiListItem)>
`;
